libnavui-dropin/src/test/java/com/mapbox/navigation/dropin/map/MapBinderTest.kt

```diff
@@ -0,0 +1,357 @@
+
+    @Test
+    @Suppress("MaxLineLength")
+    fun `bind should attach GeocodingComponent only for FreeDrive DestinationPreview and RoutePreview NavigationState`() {
+        val expectation = mapOf(
+            NavigationState.FreeDrive to true,
+            NavigationState.DestinationPreview to true,
+            NavigationState.RoutePreview to true,
+            NavigationState.ActiveNavigation to false,
+            NavigationState.Arrival to false,
+        )
+        val components = sut.bind(FrameLayout(ctx))
+        components.onAttached(mapboxNavigation)
+
+        expectation.forEach { (navigationState, shouldBind) ->
+            store.updateState { it.copy(navigation = navigationState) }
+            val c = components.findComponent { it is GeocodingComponent }
+            if (shouldBind) {
+                assertNotNull("Should attach GeocodingComponent for $navigationState", c)
+            } else {
+                assertNull("Should not attach GeocodingComponent for $navigationState", c)
+            }
+        }
+    }
+
+    @Test
+    @Suppress("MaxLineLength")
+    fun `bind should reload RouteArrowComponent on arrowOptions change`() {
+        store.updateState { it.copy(navigation = NavigationState.ActiveNavigation) }
+        val components = sut.bind(FrameLayout(ctx))
+        components.onAttached(mapboxNavigation)
+
+        val firstComponent = components.findComponent { it is RouteArrowComponent }
+        navContext.applyOptionsCustomization {
+            routeArrowOptions = RouteArrowOptions.Builder(ctx)
+                .withArrowColor(Color.RED)
+                .build()
+        }
+        val secondComponent = components.findComponent { it is RouteArrowComponent }
+
+        assertNotNull(firstComponent)
+        assertNotNull(secondComponent)
+        assertNotEquals(secondComponent, firstComponent)
+    }
+
+    @Test
+    @Suppress("MaxLineLength")
+    fun `bind should attach RouteArrowComponent only for ActiveNavigation NavigationState`() {
+        val expectation = mapOf(
+            NavigationState.FreeDrive to false,
+            NavigationState.DestinationPreview to false,
+            NavigationState.RoutePreview to false,
+            NavigationState.ActiveNavigation to true,
+            NavigationState.Arrival to false,
+        )
+        val components = sut.bind(FrameLayout(ctx))
+        components.onAttached(mapboxNavigation)
+
+        expectation.forEach { (navigationState, shouldBind) ->
+            store.updateState { it.copy(navigation = navigationState) }
+            val c = components.findComponent { it is RouteArrowComponent }
+            if (shouldBind) {
+                assertNotNull("Should attach RouteArrowComponent for $navigationState", c)
+            } else {
+                assertNull("Should not attach RouteArrowComponent for $navigationState", c)
+            }
+        }
+    }
+
+    @Test
+    fun `bind should reload RouteArrowComponent on mapStyle change`() {
```

Comment:
It should also reload on routeArrowOptions change.

Reply:
I cover that in a separate test https://github.com/mapbox/mapbox-navigation-android/pull/6446/files#diff-dcc047be08034dcd81128312c2da7ea914b85cb3f77587c85afaaa3e91e35d91R294
Line 294.